awx/main/dispatch/periodic.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import threading
+import time
+
+from django.conf import settings
+from django.db import connections
+from schedule import Scheduler
+
+from awx.main.dispatch.worker import TaskWorker
+
+logger = logging.getLogger('awx.main.dispatch.periodic')
+
+
+class Scheduler(Scheduler):
+
+    def run_continuously(self):
+        cease_continuous_run = threading.Event()
+        idle_seconds = min(self.jobs).period.total_seconds()
```

Comment:
Almost. https://en.wikipedia.org/wiki/Nyquist%E2%80%93Shannon_sampling_theorem TL;DR sampling rate should be about 2*period. So divide this result by 2, round down (except if it makes it 0), and I'm good